hudi-sync/hudi-sync-common/src/main/java/org/apache/hudi/sync/common/metrics/HoodieMetaSyncMetrics.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.sync.common.metrics;
+
+import org.apache.hudi.common.util.StringUtils;
+import org.apache.hudi.common.util.VisibleForTesting;
+import org.apache.hudi.config.metrics.HoodieMetricsConfig;
+import org.apache.hudi.hadoop.fs.HadoopFSUtils;
+import org.apache.hudi.metrics.Metrics;
+import org.apache.hudi.storage.HoodieStorage;
+import org.apache.hudi.storage.HoodieStorageUtils;
+import org.apache.hudi.sync.common.HoodieSyncConfig;
+
+import com.codahale.metrics.Counter;
+import com.codahale.metrics.Timer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class HoodieMetaSyncMetrics {
+
+  private static final Logger LOG = LoggerFactory.getLogger(HoodieMetaSyncMetrics.class);
+  // Metrics are shut down by the shutdown hook added in the Metrics class
+  private Metrics metrics;
+  private HoodieMetricsConfig metricsConfig;
+  private transient HoodieStorage storage;
+
+  private final String syncToolName;
+
+  private String recreateAndSyncFailureCounterName;
+  private String recreateAndSyncTimerName;
+
+  private Timer recreateAndSyncTimer;
+  private Counter recreateAndSyncFailureCounter;
+
+  public HoodieMetaSyncMetrics(HoodieSyncConfig config, String syncToolName) {
+    this.metricsConfig = config.getMetricsConfig();
+    this.syncToolName = syncToolName;
+    if (metricsConfig.isMetricsOn()) {
+      this.storage = HoodieStorageUtils.getStorage(config.getBasePath(), HadoopFSUtils.getStorageConf(config.getHadoopConf()));
+      metrics = Metrics.getInstance(metricsConfig, storage);
+      this.recreateAndSyncTimerName = getMetricsName("timer", "meta_sync.recreate_table");
```

Comment:
Let's define static final String for these constants.

Reply:
I've made them static. Can't make these vars as final, since they're only initialized when metrics are turned on.